src/Sulu/Bundle/WebsiteBundle/Routing/ContentRouteProvider.php

```diff
@@ -61,224 +59,157 @@ class ContentRouteProvider implements RouteProviderInterface
-     * @param ReplacerInterface $urlReplacer
      */
     public function __construct(
         DocumentManagerInterface $documentManager,
         DocumentInspector $documentInspector,
         ResourceLocatorStrategyPoolInterface $resourceLocatorStrategyPool,
         StructureManagerInterface $structureManager,
-        RequestAnalyzerInterface $requestAnalyzer,
-        DefaultLocaleProviderInterface $defaultLocaleProvider,
-        ReplacerInterface $urlReplacer
+        RequestAnalyzerInterface $requestAnalyzer
     ) {
         $this->documentManager = $documentManager;
         $this->documentInspector = $documentInspector;
         $this->resourceLocatorStrategyPool = $resourceLocatorStrategyPool;
         $this->structureManager = $structureManager;
         $this->requestAnalyzer = $requestAnalyzer;
-        $this->defaultLocaleProvider = $defaultLocaleProvider;
-        $this->urlReplacer = $urlReplacer;
     }
 
     /**
-     * Finds the correct route for the current request.
-     * It loads the correct data with the content mapper.
-     *
-     * @param Request $request A request against which to match
-     *
-     * @return \Symfony\Component\Routing\RouteCollection with all Routes that
-     *                                                    could potentially match $request. Empty collection if nothing can
-     *                                                    match
+     * {@inheritdoc}
      */
     public function getRouteCollectionForRequest(Request $request)
     {
         $collection = new RouteCollection();
 
+        /** @var RequestAttributes $attributes */
+        $attributes = $request->attributes->get('_sulu', new RequestAttributes());
+        $matchType = $attributes->getAttribute('matchType');
+
         // no portal information without localization supported
-        if ($this->requestAnalyzer->getCurrentLocalization() === null
-            && $this->requestAnalyzer->getMatchType() !== RequestAnalyzerInterface::MATCH_TYPE_PARTIAL
-            && $this->requestAnalyzer->getMatchType() !== RequestAnalyzerInterface::MATCH_TYPE_REDIRECT
+        if ($attributes->getAttribute('localization') === null
+            && $matchType !== RequestAnalyzerInterface::MATCH_TYPE_PARTIAL
+            && $matchType !== RequestAnalyzerInterface::MATCH_TYPE_REDIRECT
         ) {
             return $collection;
         }
 
-        $htmlExtension = '.html';
-        $resourceLocator = $this->requestAnalyzer->getResourceLocator();
+        $resourceLocator = $attributes->getAttribute('resourceLocator');
+        $prefix = $attributes->getAttribute('resourceLocatorPrefix');
 
-        if (
-            $this->requestAnalyzer->getMatchType() == RequestAnalyzerInterface::MATCH_TYPE_REDIRECT
-            || $this->requestAnalyzer->getMatchType() == RequestAnalyzerInterface::MATCH_TYPE_PARTIAL
+        if ($request->getPathInfo() !== $prefix . $resourceLocator
```

Comment:
this happens when you call a url like this `/test.html` in this case the content-route-provider should not match that url.